numpy/_core/tests/_natype.py

```diff
@@ -74,8 +74,7 @@ def __bool__(self):
         raise TypeError("boolean value of NA is ambiguous")
 
     def __hash__(self):
-        exponent = 31 if is_32bit else 61
-        return 2**exponent - 1
+        return 2**61 - 1
```

Comment:
I assume this is string specific? Might be good to have "string" in the file name somewhere.